docs/deposit.md

```diff
@@ -0,0 +1,168 @@
+  	// update pool tokens.
+  	for _, token := range msg.Tokens {
+  		pool.AddAsset(*token)
+  	}
+  } else {
+  	// switch pool status to 'READY'
+  	pool.Status = types.PoolStatus_POOL_STATUS_READY
+  }
+
+    newDepositID := len(depositLedger[msg.sender])
+    newDeposit := DepositMetadata {
+        ID: newDepositID,
+        PoolToken: poolToken,
+        ExpireDate: ctx.BlockTime(),
+    }
+
+    k.AddNewDepositToLedger(ctx, msg.sender, newDeposit)
+  ```
+
+  - update `OnSingleDepositAcknowledged`
+
+  ```
+  func (k Keeper) OnSingleDepositAcknowledged(ctx sdk.Context, req *types.MsgDepositRequest, res *types.MsgDepositResponse) error {
+    ...
+    newDepositID := len(depositLedger[req.sender])
+    newDeposit := DepositMetadata {
+        ID: newDepositID,
+        PoolToken: *res.PoolToken,
+        ExpireDate ctx.BlockTime() + msg.Tick,
+    }
+    k.AddNewDepositToLedger(ctx, msg.sender, newDeposit)
+  }
+  ```
+
+  - update withdraw
+
+  ```
+   func (k Keeper) OndWithdrawReceive(ctx sdk.Context, msg *types.MsgWithdrawRequest) (*types.MsgWithdrawResponse, error) {
+
+  	if err := msg.ValidateBasic(); err != nil {
+  		return nil, err
+  	}
+  	pool, found := k.GetInterchainLiquidityPool(ctx, msg.PoolCoin.Denom)
+
+  	if !found {
+  		return nil, types.ErrNotFoundPool
+  	}
+
+      depositLedger := GetDepositLedger(ctx, msg.sender)
+      withdrawableAmount := 0
+      startIndex := 0
+      for index, metadata := range depositLedger {
+          if metadata.ExpireDate > time.now() {
+              withdrawableAmount.add(metadata.PoolToken)
+          }else{
+              break
+          }
+          if withdrawableAmount >=  msg.Tokens[0] {
+              startIndex = index
+              remainAmount := withdrawableAmount.sub(metadata.PoolToken)
+              if remainAmount.Uint64() > sdk.NewInt(0) {
+                  depositLedger[index].Tokens = remainAmount
+                   startIndex = index
+              }else{
+                  startIndex = index+1
+              }
+              break
+          }
+      }
+
+    if withdrawableAmount < msg.Token[0] {
+        return nil, ErrNotEnoughWithdrawableTokens
+        }
+
+      if startIndex != 0 {
+          updatedDepositLedger = depositLedger[startIndex:len(depositLedger)]
+          k.SetDepositLedger(ctx, msg.sender, updatedDepositLedger)
+      }
+      ...
+  }
+  ```
+
+Add the same update logic `OnWithdrawAcknowledged`.
+
+##### Note
+
+- In this case, users obtain their LP tokens immediately within their wallets.
+- However, there is a loop to calculate the withdrawable amount. If the ledger continues to grow, it may impact the chain's performance. Implementing a limitation on the number of deposits may be a better approach. We can assume that a user is a malicious actor if they do not withdraw at all and only attempt to deposit multiple times. Additionally, we should establish a minimum deposit amount to prevent attacks.
```

Comment:
```
If the ledger continues to grow, it may impact the chain's performance.
```
Yeah. It would be difficult to maintain a large ledger.  Let's do a research on other AMM DEX,  this is a common issue for all LP

Reply:
Yes. If we provide three options, in this case,  calculation complexity will be increase because Deposit date don't sorted according to the expire date. 
To avoid this problem, there is a room to update data structure. for example with Heap, can resolve this problem. 
In this case, it will be same waste case calculation complexity but will improve average calculation complexity. 

But it's really not good to manage large ledger in protocol, that's why I'm suggesting second second plan. 
instead of NFT, maybe we can mint special token based on denom. Denom will include deposit metadata. 
In this case,  there is no need to include NFT module